src/Marten/IRetryPolicy.cs

```diff
@@ -66,4 +67,118 @@ public Task<TResult> ExecuteAsync<TResult>(Func<Task<TResult>> operation, Cancel
+        private static readonly Func<int, TimeSpan> DefaultSleep = x => TimeSpan.FromSeconds(x);
+
+        private DefaultRetryPolicy(int maxTries, Func<Exception, bool> filter, Func<int, TimeSpan> sleep)
+        {
+            _maxTries = maxTries;
+            _filter = filter ?? DefaultFilter;
+            _sleep = sleep ?? DefaultSleep;
+        }
+
+        /// <summary>
+        /// Initializes a retry policy that will retry once after failure.
+        /// </summary>
+        /// <param name="filter">Optional filter when to apply, default to checking for <see cref="NpgsqlException.IsTransient"/></param>
+        /// <param name="sleep">Optional sleep after exception, gets retry number 1-N as parameter, defaults to sleeping retry number seconds</param>
+        /// <returns>The configured retry policy.</returns>
+        public static IRetryPolicy Once(Func<Exception, bool> filter = null, Func<int, TimeSpan> sleep = null)
+        {
+            return new DefaultRetryPolicy(2, filter, sleep);
+        }
+
+        /// <summary>
+        /// Initializes a retry policy that will retry twice after failure.
+        /// </summary>
+        /// <param name="filter">Optional filter when to apply, default to checking for <see cref="NpgsqlException.IsTransient"/></param>
+        /// <param name="sleep">Optional sleep after exception, gets retry number 1-N as parameter, defaults to sleeping retry number seconds</param>
+        /// <returns>The configured retry policy.</returns>
+        public static IRetryPolicy Twice(Func<Exception, bool> filter = null, Func<int, TimeSpan> sleep = null)
+        {
+            return new DefaultRetryPolicy(3, filter, sleep);
+        }
+
+        /// <summary>
+        /// Initializes a retry policy that will retry given amount of times after failure.
+        /// </summary>
+        /// <param name="filter">Optional filter when to apply, default to checking for <see cref="NpgsqlException.IsTransient"/></param>
+        /// <param name="sleep">Optional sleep after exception, gets retry number 1-N as parameter, defaults to sleeping retry number seconds</param>
+        /// <returns>The configured retry policy.</returns>
+        public static IRetryPolicy Times(int times, Func<Exception, bool> filter = null, Func<int, TimeSpan> sleep = null)
+        {
+            return new DefaultRetryPolicy(times + 1, filter, sleep);
+        }
+
+        void IRetryPolicy.Execute(Action operation)
+        {
+            using (Util.NoSynchronizationContextScope.Enter())
+            {
+                Try(() =>
+                {
+                    operation();
+                    return Task.CompletedTask;
+                }, CancellationToken.None).GetAwaiter().GetResult();
+            }
+        }
+
+        TResult IRetryPolicy.Execute<TResult>(Func<TResult> operation)
+        {
+            return Try(() => Task.FromResult(operation()), CancellationToken.None).GetAwaiter().GetResult();
```

Comment:
@lahma this one would be also good to wrap with `NoSynchronizationContextScope`.

Reply:
Ah, missed that one, will amend to earlier commit.